operators/test/e2e/cmd/run/command.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package run
+
+import (
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type runFlags struct {
+	managedNamespaces   []string
+	e2eImage            string
+	elasticStackVersion string
+	kubeConfig          string
+	operatorImage       string
+	testContextOutPath  string
+	testLicence         string
+	scratchDirRoot      string
+	testRegex           string
+	testRunName         string
+	commandTimeout      time.Duration
+	autoPortForwarding  bool
+	skipCleanup         bool
+	local               bool
+}
+
+var log logr.Logger
+
+func Command() *cobra.Command {
+	flags := runFlags{}
+
+	cmd := &cobra.Command{
+		Use:   "run",
+		Short: "setup an e2e test environment and run tests",
+		PreRun: func(_ *cobra.Command, _ []string) {
+			log = logf.Log.WithName(flags.testRunName)
+		},
+		RunE: func(_ *cobra.Command, _ []string) error {
+			return doRun(flags)
+		},
+	}
+
+	cmd.Flags().BoolVar(&flags.autoPortForwarding, "auto-port-forwarding", false, "Enable port forwarding to pods")
+	cmd.Flags().DurationVar(&flags.commandTimeout, "command-timeout", 90*time.Second, "Timeout for commands executed")
+	cmd.Flags().StringVar(&flags.e2eImage, "e2e-image", "", "E2E test image")
+	cmd.Flags().StringVar(&flags.elasticStackVersion, "elastic-stack-version", "7.1.1", "Elastic stack version")
+	cmd.Flags().StringVar(&flags.kubeConfig, "kubeconfig", "", "Path to kubeconfig")
+	cmd.Flags().BoolVar(&flags.local, "local", false, "Create the environment for running tests locally")
+	cmd.Flags().StringSliceVar(&flags.managedNamespaces, "managed-namespaces", []string{"mercury", "venus"}, "List of managed namespaces")
+	cmd.Flags().StringVar(&flags.operatorImage, "operator-image", "", "Operator image")
+	cmd.Flags().BoolVar(&flags.skipCleanup, "skip-cleanup", false, "Do not run cleanup actions after test run")
+	cmd.Flags().StringVar(&flags.testContextOutPath, "test-context-out", "", "Write the test context to the given path")
+	cmd.Flags().StringVar(&flags.testLicence, "test-licence", "", "Test licence to apply")
+	cmd.Flags().StringVar(&flags.scratchDirRoot, "scratch-dir", "/tmp/eck-e2e", "Path under which temporary files should be created")
+	cmd.Flags().StringVar(&flags.testRegex, "test-regex", "", "Regex to pass to the test runner")
```

Comment:
Can we pass these as a config file too? Currently we will be in the same game of passing those args from jenkins through docker and many makefile targets with every new setting we want to introduce. I think that right now it only needs test-regex and operator image, but more are likely to follow (es version? e2e image?)

Reply:
Currently there's only a single target in the Makefile where these flags are set so there's no repetition happening. The issue with using a static config file is that some parameters such as the operator image name is dynamic (changes according to the user and commit). It would be  annoying to edit the config file on every commit to match the current HEAD whereas passing it as a flag or an environment variable is much easier.

Reply:
That single target is the entry to e2e tests - while it's a single place, it's going to be invoked on multiple paths. And I'm not proposing to use a static config file and edit it with every commit:) 

Let's say we want to run nightly against multiple es versions - how we can achieve that? It seems that right now we'd need to introduce the variable in jenkins job, pass it to the container, pass it in makefile and set the default somewhere. I'd prefer to avoid editing three places to add a single test variable. And this issue will only get amplified when we add more dimensions and we keep stacking args lists on every level.

I'm fine with doing it in a separate PR (as this one is big as it is), but I do feel that this would be useful right away.

Reply:
Makes sense. I'll consider doing that in a separate PR.